zkvm/src/predicate.rs

```diff
@@ -21,25 +21,23 @@ pub enum Predicate {
     /// Verifier's view on the predicate in a compressed form to defer decompression cost.
     Opaque(CompressedRistretto),
 
-    /// Prover's view on the predicate that has valid representation.
-    Witness(PredicateWitness),
-}
-
-/// Prover's representation of the predicate tree with all the secret witness data.
-#[derive(Clone, Debug)]
-pub enum PredicateWitness {
-    /// Representation of an opaque branch of the predicate tree
-    /// (but known to be a valid Ristretto point).
-    OpaqueBranch(RistrettoPoint),
-
-    /// Secret signing key for the predicate-as-a-verification-key.
-    Key(Scalar),
+    /// Signing key for the predicate-as-a-verification-key.
+    /// Prover will provide secret signing key separately when
+    /// constructing aggregated signature.
+    Key(VerificationKey),
 
     /// Representation of a predicate as commitment to a program.
     Program(Vec<Instruction>),
 
     /// Disjunction of two predicates.
-    Or(Box<PredicateWitness>, Box<PredicateWitness>),
+    Or(PredicateDisjunction),
```

Comment:
Maybe box things higher in the hierarchy, so that fewer types have indirect storage?

```rust
pub enum Predicate {
    ...
    Or(Box<PredicateDisjunction>)
}

pub struct PredicateDisjunction {
     left: Predicate,
     right: Predicate,
     point: CompressedRistretto,
}
```

WDYT?

Reply:
ah, that's a great suggestion! thanks :)